src/main/java/com/vackosar/gitflowincrementalbuild/boundary/MavenLifecycleParticipant.java

```diff
@@ -51,10 +53,14 @@ public void afterProjectsRead(MavenSession session) throws MavenExecutionExcepti
     }
 
     private void execute(MavenSession session) throws GitAPIException, IOException {
-        Guice
-                .createInjector(new GuiceModule(logger, session))
-                .getInstance(UnchangedProjectsRemover.class)
-                .act();
+        Injector injector = Guice.createInjector(new GuiceModule(logger, session));
+        try {
+            injector.getInstance(UnchangedProjectsRemover.class).act();
+        } finally {
```

Comment:
A note regarding this change which is linked to the removal of `close()` calls in `DifferentFiles`:
The repo should always be closed and closing should happen in the class that effectively opened it.
Tha's why I moved this here into a `finally` block.